### PR TITLE
(PC-33458) feat(AvatarList): add new Avatar List

### PR DESCRIPTION
**Pull request by yleclercq-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-33458

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [x] Made sure my feature is working on web.
- [x] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]
- [ ] I am aware of all the best practices and respected them.

## Screenshots

**delete** _if no UI change_

Capture d’écran 2024-12-10 à 14 48 51
Capture d’écran 2024-12-11 à 12 19 20

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a
[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4

## Best Practices

<details>
  <summary>Click to expand</summary>
These rules apply to files that you make changes to.
If you can't respect one of these rules, be sure to explain why with a comment.
If you consider correcting the issue is too time consuming/complex: create a ticket. Link the ticket in the code.

- In the production code: remove type assertions with `as` (type assertions are removed at compile-time, there is no runtime checking associated with a type assertion. There won’t be an exception or `null` generated if the type assertion is wrong). In certain cases `as const` is acceptable (for example when defining readonly arrays/objects). Using `as` in tests is tolerable.
- Remove bypass type checking with `any` (when you want to accept anything because you will be blindly passing it through without interacting with it, you can use `unknown`). Using `any` in tests is tolerable.
- Remove non-null assertion operators (just like other type assertions, this doesn’t change the runtime behavior of your code, so it’s important to only use `!` when you know that the value can’t be `null` or `undefined`).
- Remove all `@ts-expect-error` and `@eslint-disable`.
- Remove all warnings, and errors that we are used to ignore (`yarn test:lint`, `yarn test:types`, `yarn start:web`...).
- Use `gap` (`ViewGap`) instead of `<Spacer.Column />`, `<Spacer.Row />` or `<Spacer.Flex />`.
- Don't add new "alias hooks" (hooks created to group other hooks together). When adding new logic, this hook will progressively become more complex and harder to maintain.
- Remove logic from components that should be dumb.

Test specific:

- Avoid mocking internal parts of our code. Ideally, mock only external calls.
- When you see a local variable that is over-written in every test, mock it.
- Prefer `user` to `fireEvent`.
- When mocking feature flags, use `setFeatureFlags`. If not possible, mention which one(s) you want to mock in a comment (example: `jest.spyOn(useFeatureFlagAPI, 'useFeatureFlag').mockReturnValue(true) // WIP_NEW_OFFER_TILE in renderPassPlaylist.tsx` )
- In component tests, replace `await act(async () => {})` and `await waitFor(/* ... */)` by `await screen.findBySomething()`.
- In hooks tests, use `act` by default and `waitFor` as a last resort.
- Make a snapshot test for pages and modals ONLY.
- Make a web specific snapshot when your web page/modal is specific to the web.
- Make an a11y test for web pages.

Advice:

- Use TDD
- Use Storybook
- Use pair programming/mobs
</details>

### CODE REVIEW

**Comment by yleclercq-pass** [comment]
AvatarList sera testé lors de l'implémentation avec de la vrai data sauf si vous voulez que je le teste sur cette PR mais les tests devront être modifié

File: src/ui/components/Avatar/AvatarList.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// Remove Fake Data
+const avatarsData = [
+  { id: 1, image: <Profile size={40} testID="profil" />, name: 'Oda' },
+  { id: 2, image: <TypoDS.Title1>M.M</TypoDS.Title1>, name: 'MMMM' },
+  { id: 3, name: 'Lolo' },
+]
+
+const GAP = getSpacing(2)
+const PLAYLIST_ITEM_HEIGHT =
+  AVATAR_LARGE + Number(theme.designSystem.typography.title4.fontSize) + GAP
```
Comment (clesausse-pass) [inline src/ui/components/Avatar/AvatarList.tsx:20]: Il ne vaut mieux pas que tu utilises parseFloat à cause du px ici ?

**Review by clesausse-pass — APPROVED** [review]

File: src/ui/components/Avatar/AvatarList.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+]
+
+const GAP = getSpacing(2)
+const PLAYLIST_ITEM_HEIGHT =
+  AVATAR_LARGE + parseFloat(theme.designSystem.typography.title4.fontSize) + GAP
+const PLAYLIST_ITEM_WIDTH = AVATAR_LARGE
+
+export const AvatarsList = () => {
+  const renderAvatar = ({ item }: { item: (typeof avatarsData)[0] }) => (
+    <AvatarListItem image={item.image} name={item.name} />
```
Comment (mmeissonnier-pass) [inline src/ui/components/Avatar/AvatarList.tsx:25]: Je ne sais plus si on doit aussi pouvoir naviguer sur la page artiste. Mais si c'est le cas  il faudrait wrapper ton Item dans un Touchable et faire un callback `onItemPress` qui passe l'id en param

File: src/ui/components/Avatar/AvatarList.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import { theme } from 'theme'
+import { AvatarListItem } from 'ui/components/Avatar/AvatarListItem'
+import { Playlist } from 'ui/components/Playlist'
+import { Profile } from 'ui/svg/icons/Profile'
+import { getSpacing, TypoDS } from 'ui/theme'
+import { AVATAR_LARGE } from 'ui/theme/constants'
+
+// Remove Fake Data
+const avatarsData = [
```
[the remark this answers is not in the document]
Comment (yleclercq-pass) [reply src/ui/components/Avatar/AvatarList.tsx:12]: C'est ce que je voulais avec la vrai la data du coup je le fais avec la fake :)

File: src/ui/components/Avatar/DefaultAvatar.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import LinearGradient from 'react-native-linear-gradient'
+import styled from 'styled-components/native'
+
+import { BicolorProfile } from 'ui/svg/icons/BicolorProfile'
+
+export const DefaultAvatar = styled(LinearGradient).attrs(({ theme }) => ({
```
Comment (bpeyrou-pass) [inline src/ui/components/Avatar/DefaultAvatar.tsx:7]: J'ai vu qu'il y avait un DefaultAvatar dans la stories de Avatar, c'est mergeable du coup ?

**Review by mmeissonnier-pass — APPROVED** [review]

File: src/ui/components/Avatar/AvatarListItem.native.test.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+import { render, screen } from 'tests/utils'
+import { AvatarListItem } from 'ui/components/Avatar/AvatarListItem'
+import { Profile } from 'ui/svg/icons/Profile'
+
+describe('<AvatarListItem />', () => {
+  it('should display artist avatar name', () => {
+    render(<AvatarListItem name="Oda" />)
```
Comment (bpeyrou-pass) [inline src/ui/components/Avatar/AvatarListItem.native.test.tsx:9]: t'as des erreurs de lint

**Review by bpeyrou-pass — APPROVED** [review]
une fois le linter fix c'est good pour moi

**Pull request merged**